data-node/src/main/java/org/graylog/datanode/configuration/variants/OpensearchSecurityConfiguration.java

```diff
@@ -133,6 +144,29 @@ public Map<String, String> getProperties() throws GeneralSecurityException, IOEx
         return config.build();
     }
 
+    private Map<String, Object> getMap(final Map<String, Object> map, final String... keys) {
+        Map<String, Object> result = map;
+        for(final String key: List.of(keys)) {
+            result = (Map<String, Object>)result.get(key);
+        };
+        return result;
+    }
+
+    private void enableJwtInConfig(final Path opensearchConfigDir, final byte[] signingKey) throws IOException {
+        final File file = opensearchConfigDir.resolve(Path.of("opensearch-security", "config.yml")).toFile();
```

Comment:
Should we wrap the `File` reference in a try-with-resource block, to make sure it gets closed when the method ends?
```suggestion
        try (final File file = opensearchConfigDir.resolve(Path.of("opensearch-security", "config.yml")).toFile()) {
```

Reply:
File is not autocloseable, I guess `objectMapper.readValue(` has to take care of it?